gtsam/base/numericalDerivative.h

```diff
@@ -19,20 +19,15 @@
 #pragma once
 
 #include <boost/function.hpp>
-#ifdef __GNUC__
```

Comment:
Any reason this was removed?

Reply:
This was removed as I believe it was first introduced to surpress warnings from the previous `<boost/bind.hpp>` header. As we are not using that header, we can remove this `ifdef`.

Reply:
Even if we do use the older include, I believe the later versions of boost have fixed the associated warnings so there is no need to suppress them.